### PR TITLE
Add tools container support

### LINKED ISSUE #1176: Have a reccomended CLI-Tools distribution for most platforms
**Is your feature request related to a problem? Please describe.**
As it stands the only reccomended way to get the cli tools is to use opensuse tumbleweed.
Many people might think of docker, but perhaps not all, and in any case giving your reccomendation for some way to have client tools would make entry easier and provide validation of it working by virtue of prople doing it and raising issues if it doesnt work.

Thats not put well, but I'm tired ;-P.

**Describe the solution you'd like**
Either clear up the packaging situation which seems difficult, or provide a cli container, or at least reccomend that one use the opensuse/tumbleweed container to install the tools.

**Describe alternatives you've considered**
cant think of anything rn

**Additional context**
-

### PR DESCRIPTION
**Pull request by Firstyear**
Fixes #1176 - Add a tools container and improve some docs adjacent to tools install. I intend to backport this to the .10 release. 

- [ x ] cargo fmt has been run
- [ ] cargo clippy has been run
- [ x ] cargo test has been run and passes
- [ ] book chapter included (if relevant)
- [ ] design document included (if relevant)

### CODE REVIEW

File: kanidm_book/src/installing_client_tools.md [hunk 1]
```diff
@@ -38,21 +37,44 @@ Then you need to refresh your metadata and install the clients.
...
+
+    docker pull kanidm/tools:latest
+    docker run --rm -i -t \
+        -v /etc/kanidm/config:/etc/kanidm/config:ro \
+        -v ~/.config/kanidm:/home/kanidm/.config/kanidm:ro \
+        -v ~/.cache/kanidm_tokens:/home/kanidm/.cache/kanidm_tokens \
+        kanidm/tools:latest \
+        /sbin/kanidm --help
+
+If you have a ca.pem you may need to bind mound this in as requried.
```
Comment (yaleman) [inline kanidm_book/src/installing_client_tools.md:72]:
```suggestion
If you have a ca.pem you may need to bind mount this in as required.
```

File: kanidm_client/src/lib.rs [hunk 2]
```diff
@@ -208,6 +208,12 @@ impl KanidmClientBuilder {
         config_path: P,
     ) -> Result<Self, ()> {
         debug!("Attempting to load configuration from {:#?}", &config_path);
+
```
Comment (yaleman) [inline kanidm_client/src/lib.rs:211]: "Config file not found" is already covered below this?

**Review by yaleman — CHANGES_REQUESTED** [review]

File: kanidm_client/src/lib.rs (hunk 2 above)
Comment (Firstyear) [reply kanidm_client/src/lib.rs:211]: Ahahah there was a special-hell error here where when the file doesn't exist, docker mounts were passing it through as a directory and it caused some ridiculous errors like "can't read file, this is a directory". that's why I added the extra check. I might comment that in the code.

Comment (yaleman) [reply kanidm_client/src/lib.rs:211]: Ah, I *hate* that "feature" in container-land.

**Review by yaleman — APPROVED** [review]

**Pull request merged**